### DISCUSSION TITLE
Migrate state.getStrings() to v5

### DISCUSSION THREAD

**Discussion by apostopher** [statelyai/xstate#4751] [General] [answered]
Hello All,
We are using xstate v4 for many features in our project. I'm in a process of migrating xstate v4 code to v5. I'm using `state.getStrings()` in my current implementation. i see that this method is not available in v5. can someone help me to migrate my v4 code?

**Comment by apostopher** [comment, marked as answer]
i copied the `getStrings` implementation from v4 source code.

**Comment by davidkpiano** [comment]
For convenience, I added the implementation in the migration guide: https://stately.ai/docs/migration#the-stategetstrings-method-has-been-removed

```ts
import { type StateValue } from 'xstate';

export function getStateValueStrings(stateValue: StateValue): string[] {
  if (typeof stateValue === 'string') {
    return [stateValue];
  }
  const valueKeys = Object.keys(stateValue);

  return valueKeys.concat(
    ...valueKeys.map((key) =>
      getStateValueStrings(stateValue[key], delimiter).map(
        (s) => key + delimiter + s
      )
    )
  );
}

// ...

const stateValueStrings = getStateValueStrings(stateValue);
// e.g. ['green', 'yellow', 'red', 'red.walk', 'red.wait', …]
```
